### DISCUSSION TITLE
How SSL certificate verification works

### DISCUSSION THREAD

**Discussion by Nardol** [BearWare/TeamTalk5#1706] [Q&A]
Hello,

I've compiled the QT client and pro server from master branch.
The actual server certificate is a self-signed one, so nothing can be more invalid except an expired certificate.

But I can connect to my server without any issue.

I thought it was for checking the host name, expiration and the §CA if it is legitimate.

How does it work?
Or did I miss-understand when reading the git log about certificate verification?

**Comment by bear101** [comment]
The client part of certificate verification is not yet implemented. Both client and server should use the same CA in order to do certificate verification.

**Reply by Nardol** [reply]
So it is certificate authentication (for account), not to authenticate server information as it would be the case for a HTTPS connection so only server?

**Reply by bear101** [reply]
Certificate verification will be based on the properties defined here:
https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalk_DLL/TeamTalk.h#L2638
